City-of-Helsinki/tilavarauspalvelu-core
# TPREK Unit auto-update

## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Rewrite Unit importing from TPREK
- Fetch Hauki resources for units imported from TPREK _(previously this code was never run)_
- Add task for updating Units from TPREK

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests
- In Django admin:
  - Set a `tprek_id` for any Unit
  - Run `Update from TPREK` for the Unit (from List or Edit page)
  - Check server logs for results
  - Unit information should be now updated, and a Hauki resource set (if one was found in the Hauki API)

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- #1178 (Not really needed, but still kindof needed for tests to pass :D)

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-2868][TILA]


[TILA]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-2868


[TILA-2868]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-2868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
